shared/src/main/scala/crystal/package.scala

```diff
@@ -36,3 +38,11 @@ object throwable {
         (x ne null) == (y ne null)
     }
 }
+
+given [F[_]: Monad]: Semigroupal[ViewF[F, *]] with
+  def product[A, B](fa: ViewF[F, A], fb: ViewF[F, B]): ViewF[F, (A, B)] =
+    ViewF.apply(fa.get -> fb.get, (f, cb) => cb(f(fa.get, fb.get)))
```

Comment:
The mod function is not applied to underlying `View`s. It is only applied to the callback value.

Furthermore, I suggest not using `get` in the modifier block in case the current value is stale. Instead, we should attempt to pass the mod function to the underlying layer.

In this case, we can achieve an effectful get by using the trick of a mod(identity).

Suggested change: 
```scala
def product[A, B](fa: ViewF[F, A], fb: ViewF[F, B]): ViewF[F, (A, B)] =
  ViewF.apply(
    fa.get -> fb.get,
    (f, cb) =>
      fa.modCB( // Get current a value
        identity(_),
        oldA =>
          fb.modCB( // Get current b value
            identity(_),
            oldB =>
              fa.modCB( // Modify underlying a value
                a => f(a, oldB)._1,
                newA =>
                  fb.modCB( // Modify underlying b value
                    b => f(oldA, b)._2,
                    newB => cb((newA, newB))
                  )
              )
          )
      )
  )
```